include/libdnf5/base/transaction.hpp

```diff
@@ -193,12 +194,17 @@ class LIBDNF_API Transaction {
     void set_download_local_pkgs(bool value);
     bool get_download_local_pkgs() const noexcept;
 
+    /// Retrieve output of the last executed rpm scriptlet.
+    std::string get_last_script_output();
```

Comment:
I'm wondering if we need to make a copy in the future or if it would be better to just return a constant reference.
`const std::string & get_last_script_output();`
If you change here to a reference, change in `transaction_impl.hpp` as well

Reply:
Yes, I intended to make a copy here. The thing is that this string is regularly changed from another thread and I wanted to ensure that the string the client received will remain the same.

Reply:
If the user `calls get_last_script_output` in a `script_error` or `script_stop` callback, the script has already finished and a new script will be started after the next `script_start` callback. The rpm executes scripts serially (there is no parallelization). So the contents of the string should not change while the `script_error` and `script_stop` callbacks are being serviced. And if the user needs to keep it for longer, he will make his own copy. 

Your solution allows the user to call the `get_last_script_output` method from another thread. We don't generally support multithreading - it is not supported by the libraries used, for example libsolv. So I still don't know if this copy will help us in the future. But I will merge it.

Reply:
I was specifically thinking about situations where the client keeps the output longer then during script_start/script_end execution. You are right that it can be handled on the client side. Thank you.